File: benchmark/comparison/results/blind_py/30.py
class Ok:
    def __init__(self, v):
        self.v = v
    def __repr__(self):
        return f"Ok({self.v})"

class Err:
    def __init__(self, msg):
        self.msg = msg
    def __repr__(self):
        return f"Err({self.msg})"

def validate_age(age):
    if age < 0:
        return Err("age must be non-negative")
    if age >= 150:
        return Err("age must be under 150")
    return Ok(age)

def validate_name(name):
    if name == "":
        return Err("name must not be empty")
    return Ok(name)

def validate_user(name, age):
    r_name = validate_name(name)
    if isinstance(r_name, Err):
        return r_name
    r_age = validate_age(age)
    if isinstance(r_age, Err):
        return r_age
    return Ok(f"{name} (age {age})")

print(validate_age(25))
print(validate_age(-1))
print(validate_age(200))
print(validate_name("Alice"))
print(validate_name(""))
print(validate_user("Alice", 25))
print(validate_user("", 25))
print(validate_user("Alice", -1))

File: benchmark/comparison/results/blind_ts/30.ts
type Result = { tag: "Ok"; value: string } | { tag: "Err"; msg: string };

function Ok(value: string): Result { return { tag: "Ok", value }; }
function Err(msg: string): Result { return { tag: "Err", msg }; }

function formatResult(r: Result): string {
  if (r.tag === "Ok") return `Ok(${r.value})`;
  return `Err(${r.msg})`;
}

function validate_age(age: number): Result {
  if (age < 0) return Err("age must be non-negative");
  if (age >= 150) return Err("age must be under 150");
  return Ok(String(age));
}

function validate_name(name: string): Result {
  if (name === "") return Err("name must not be empty");
  return Ok(name);
}

function validate_user(name: string, age: number): Result {
  const nameResult = validate_name(name);
  if (nameResult.tag === "Err") return nameResult;
  const ageResult = validate_age(age);
  if (ageResult.tag === "Err") return ageResult;
  return Ok(`${name} (age ${age})`);
}

console.log(formatResult(validate_age(25)));
console.log(formatResult(validate_age(-1)));
console.log(formatResult(validate_age(200)));
console.log(formatResult(validate_name("Alice")));
console.log(formatResult(validate_name("")));
console.log(formatResult(validate_user("Alice", 25)));
console.log(formatResult(validate_user("", 25)));
console.log(formatResult(validate_user("Alice", -1)));
